### PR TITLE
[UnitaryHACK] ZNE-PEC uniformity

### LINKED ISSUE #638: Make mitigate_executor and decorators more uniform between ZNE and PEC
Issue Description
-----------------
For zero noise extrapolation there are 3 alternatives ways to use Mitiq: `execute_with_zne`, `mitigate_executor` and `zne_decorator`. 

For probabilistic error cancellation, we only have `execute_with_pec`.

It would be good to have a more uniform interface. This is also relevant for the arXiv resubmission.

Proposed Solution
-----------------
I would propose two options:

1. Add `mitigate_executor` and `pec_decorator` in `mitiq.pec`.
2. Remove the `zne_decorator` from `mitiq.zne` and only add `mitigate_executor` in `mitiq.pec`. The decorator is nice but in practice is not very useful. Indeed one typically wants to keep both the mitigated and the unmitigated executor in order to compare the results, but this is not possible with a decorator.

### PR DESCRIPTION
**Pull request by andre-a-alves**
Description
-----------
Closes #638 
My contribution to the documentation was only about two sentences, because PEC seems to only be lightly mentioned. Also, my understanding of PEC is pretty low, so someone who understands it a little better may want to review the docstrings to make sure what I wrote was accurate; it was the best I could do.

Checklist
-----------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [X] I added unit tests for new code.
- [X] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [X] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [X] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.

If some items remain, you can mark this a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

Tips
----

- If the validation check fails:

    1. Run `make check-style` (from the root directory
  of the repository) and fix any [flake8](http://flake8.pycqa.org) errors.

    2. Run `make format` to format your code with the [black](https://black.readthedocs.io/en/stable/index.html)
  autoformatter.

  For more information, check the [style guidelines](https://mitiq.readthedocs.io/en/stable/contributing.html#style-guidelines) for Mitiq.
  
- Write "Fixes #XYZ" in the description if this PR fixes Issue #XYZ.

### CODE REVIEW

**Review by andreamari — CHANGES_REQUESTED** [review]
Thanks @andre-a-alves !  
I think this is a great PR. I only left one comment about import errors/warnings

File: mitiq/__init__.py [hunk 1]
```diff
@@ -17,4 +17,3 @@
 from mitiq._typing import QPROGRAM
 from mitiq._version import __version__
 from mitiq.collector import generate_collected_executor
-from mitiq.zne import execute_with_zne, mitigate_executor, zne_decorator
```
[the remark this answers is not in the document]
Comment (andreamari) [reply mitiq/__init__.py:20]: Thanks @andre-a-alves! It's not urgent, you can take all the time you need.

[the remark this answers is not in the document]
Comment (andre-a-alves) [reply mitiq/__init__.py:20]: With PendingDeprecationWarning not being displayed by default, I think raising an ImportError in the future was going to be necessary, anyway, so I just went ahead and raised it now. This solution also sets a foundation from which any future import statement changes can easily be captured in the same way, including tests for raising the errors.

**Comment by andre-a-alves** [comment]
Regarding the failed doctest... I ran the test on my computer and got no errors. Then I pushed, and these errors surfaced. At that point, they surfaced on my computer, too. About 20 minutes later, with nothing having changed, they are now passing on my computer. Is that common? And is there a way to initialize the test again for this PR?

**Comment by andreamari** [comment]
> Regarding the failed doctest... I ran the test on my computer and got no errors. Then I pushed, and these errors surfaced. At that point, they surfaced on my computer, too. About 20 minutes later, with nothing having changed, they are now passing on my computer. Is that common? And is there a way to initialize the test again for this PR?

I re-executed the tests again and got the same error. The failing test is unrelated to this PR and will be probably fixed by #671. So I think the best strategy is merging this PR after #671.

P.s. thanks for the new import errors and tests.

**Comment by rmlarose** [comment]
> Regarding the failed doctest... I ran the test on my computer and got no errors. Then I pushed, and these errors surfaced. At that point, they surfaced on my computer, too. About 20 minutes later, with nothing having changed, they are now passing on my computer. Is that common? And is there a way to initialize the test again for this PR?

Merge master to fix this now!

**Comment by andreamari** [comment]
@andre-a-alves, you probably need to merge master into this branch (especially after the recent merging of #676).
Hopefully this will fix the failing doc test.

If you are participating in [unitaryHACK](https://unitaryfund.github.io/unitaryhack/), you can label this PR by writing  `[unitaryHACK]` in the title.

**Comment by andre-a-alves** [comment]
Ironically, while it's passing here, I am now getting a doctest error on my machine (with a new environment that was installed from scratch, and on both this branch and on main).

**Comment by andreamari** [comment]
> Ironically, while it's passing here, I am now getting a doctest error on my machine (with a new environment that was installed from scratch, and on both this branch and on main).

Did you try to re-run `pip install -r dev_requirements` after merging? If you still get the problem, feel free to open a new issue with more info about the local error. Meanwhile, since CI tests are passing, we can merge this PR.

Thanks again and congrats for the first unitaryHACK PR related to the Mitiq project!

**Review by andreamari — APPROVED** [review]

**Comment by crazy4pi314** [comment]
@all-contributors please add @andre-a-alves for doc, test

**Pull request merged**